DependencyInjection/CmfSeoExtension.php

```diff
@@ -187,26 +188,30 @@ private function loadPhpcr($config, XmlFileLoader $loader, ContainerBuilder $con
      */
     private function loadAlternateLocaleProvider($config, ContainerBuilder $container)
     {
-
-        $alternateLocaleProvider = empty($config['provider_id'])
+        $alternateLocaleProvider = null === $config['provider_id']
             ? $this->defaultAlternateLocaleProviderId
             : $config['provider_id'];
 
 
         if ($alternateLocaleProvider) {
-            $alternateLocaleProviderDefinition = $container->findDefinition($alternateLocaleProvider);
-            $container
-                ->findDefinition('cmf_seo.event_listener.seo_content')
-                ->addMethodCall(
-                    'setAlternateLocaleProvider',
-                    array($alternateLocaleProviderDefinition)
-                );
-            $container
-                ->findDefinition('cmf_seo.sitemap.phpcr_provider')
-                ->addMethodCall(
-                    'setAlternateLocaleProvider',
-                    array($alternateLocaleProviderDefinition)
-                );
+            try {
+                $alternateLocaleProviderDefinition = $container->findDefinition($alternateLocaleProvider);
+                $container
+                    ->findDefinition('cmf_seo.event_listener.seo_content')
+                    ->addMethodCall(
+                        'setAlternateLocaleProvider',
+                        array($alternateLocaleProviderDefinition)
+                    );
+                $container
+                    ->findDefinition('cmf_seo.sitemap.guesser')
+                    ->addMethodCall(
+                        'setAlternateLocaleProvider',
+                        array($alternateLocaleProviderDefinition)
+                    );
+            } catch(InvalidArgumentException $e) {
```

Comment:
but then we should do $container->hasDefinition or whatever its called to check, if we can't even set a flag in the class to know whether the guesser should be available or not. silently discarding an exception is quite risky, and should not be needed for this case.